iocore/eventsystem/I_VConnection.h

```diff
@@ -362,7 +362,13 @@ class VConnection : public Continuation
     return false;
   }
 
-public:
+  // This function should be called when the VConnection is a tunnel endpoint.  By default, a VConnection does not care if it
+  // is a tunnel endpoint.
+  virtual void
+  mark_as_tunnel_endpoint()
```

Comment:
Why do we need this virtual function in the `VConnection` class? As you commented, `VConnection` doesn't care about it's tunnel or something else.

Reply:
Because it's called through a VConnection pointer at certain places.

Reply:
This one?
```
HttpSM::setup_blind_tunnel(bool send_response_hdr, IOBufferReader *initial) {
...
   ua_entry->vc->mark_as_tunnel_endpoint();
   server_entry->vc->mark_as_tunnel_endpoint();
```

Can we use `ua_txn->get_netvc()` and `server_txn->get_netvc()` instead of `ua_entry->vc` and `server_entry->vc` here?

Reply:
OK I just tried that.  `server_txn` is null at that point in the code, so it doesn't work.

Reply:
Fair enough, thanks for checking.